examples/cmd/vppagent-icmp-responder-nse/nse.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"sync"
+	"syscall"
+
+	"github.com/ligato/networkservicemesh/controlplane/pkg/nsmd"
+
+	"github.com/ligato/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/ligato/networkservicemesh/controlplane/pkg/apis/registry"
+
+	"github.com/ligato/networkservicemesh/pkg/tools"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+const (
+	// networkServiceName defines Network Service Name the NSE is serving for
+	NetworkServiceName      = "vppagent-icmp-responder"
+	DefaultVPPAgentEndpoint = "localhost:9112"
+)
+
+func main() {
+	var wg sync.WaitGroup
+
+	nsmServerSocket, _ := os.LookupEnv(nsmd.NsmServerSocketEnv)
+	logrus.Infof("nsmServerSocket: %s", nsmServerSocket)
+	// TODO handle missing env
+
+	nsmClientSocket, _ := os.LookupEnv(nsmd.NsmClientSocketEnv)
+	logrus.Infof("nsmClientSocket: %s", nsmClientSocket)
+	// TODO handle missing env
+
+	workspace, _ := os.LookupEnv(nsmd.WorkspaceEnv)
+	logrus.Infof("workspace: %s", workspace)
+	// TODO handle missing env
+
+	// For NSE to program container's dataplane, container's linux namespace must be sent to NSM
+	linuxNS, err := tools.GetCurrentNS()
+	if err != nil {
+		logrus.Fatalf("nse: failed to get a linux namespace with error: %v, exiting...", err)
+		os.Exit(1)
+	}
+	logrus.Infof("Starting NSE, linux namespace: %s", linuxNS)
+
+	if err := tools.SocketCleanup(nsmClientSocket); err != nil {
+		logrus.Fatalf("nse: failure to cleanup stale socket %s with error: %v", nsmClientSocket, err)
+	}
+
+	logrus.Infof("nse: listening socket %s", nsmClientSocket)
+	connectionServer, err := net.Listen("unix", nsmClientSocket)
+	if err != nil {
+		logrus.Fatalf("nse: failure to listen on a socket %s with error: %v", nsmClientSocket, err)
+	}
+	grpcServer := grpc.NewServer()
+
+	// Registering NSE API, it will listen for Connection requests from NSM and return information
+	// needed for NSE's dataplane programming.
+	nseConn := New(DefaultVPPAgentEndpoint, path.Join("/var/lib/networkservicemesh", workspace))
```

Comment:
I'd suggest here that rather than a join, you simply have the nsmdp pass through a Workspace env variable (which will usually simply point to "/var/lib/networkservicemesh" and just use that here :)

Reply:
Yes, it makes sense :)